src/app/core/components/widgets/widget-network/widget-network.component.ts

```diff
@@ -0,0 +1,349 @@
+  cols = 2;
+  rows = 2;
+  gap = 16;
+  contentHeight = 400 - 56;
+  rowHeight = 150;
+  aspectRatio = 474 / 200;
+
+  chartData: ChartData = {};
+  interval: Interval;
+  availableNics: BaseNetworkInterface[] = [];
+  chartOptions: ChartOptions = {
+    responsive: true,
+    maintainAspectRatio: true,
+    aspectRatio: this.aspectRatio,
+    layout: {
+      padding: 0,
+    },
+    legend: {
+      align: 'end',
+      labels: {
+        boxWidth: 8,
+        usePointStyle: true,
+      },
+    },
+    scales: {
+      xAxes: [
+        {
+          type: 'time',
+          time: {
+            unit: 'minute',
+            displayFormats: {
+              minute: 'HH:mm',
+            },
+          },
+          ticks: {
+            beginAtZero: true,
+            maxTicksLimit: 3,
+            maxRotation: 0,
+          },
+        },
+      ],
+      yAxes: [
+        {
+          position: 'right',
+          ticks: {
+            maxTicksLimit: 8,
+            callback: (value) => {
+              return this.convertKMGT(value);
+            },
+          },
+        },
+      ],
+    },
+  };
+
+  constructor(
+    public router: Router, private ws: WebSocketService, private locale: LocaleService,
+    private tableService: TableService, public translate: TranslateService,
+  ) {
+    super(translate);
+    this.configurable = false;
+    this.utils = new WidgetUtils();
+  }
+
+  ngOnDestroy(): void {
+    this.core.emit({ name: 'StatsRemoveListener', data: { name: 'NIC', obj: this } });
+    this.core.unregister({ observerClass: this });
+
+    if (this.interval) {
+      clearInterval(this.interval);
+    }
+  }
+
+  ngAfterViewInit(): void {
+    this.availableNics = this.nics.filter((nic) => nic.state.link_state === 'LINK_STATE_UP');
+
+    this.updateGridInfo();
+    this.updateMapInfo();
+
+    this.availableNics.forEach((nic) => {
+      this.fetchReportData(nic);
+    });
+
+    if (this.interval) {
+      clearInterval(this.interval);
+    }
+
+    this.interval = setInterval(() => {
+      this.availableNics.forEach((nic) => {
+        this.fetchReportData(nic);
+      });
+    }, 60000);
+
+    this.stats.pipe(untilDestroyed(this)).subscribe((evt: CoreEvent) => {
+      if (evt.name.startsWith('NetTraffic_')) {
+        const nicName = evt.name.substr('NetTraffic_'.length);
+        if (nicName in this.nicInfoMap) {
+          const sent = this.utils.convert(evt.data.sent_bytes_rate);
+          const received = this.utils.convert(evt.data.received_bytes_rate);
+
+          const nicInfo = this.nicInfoMap[nicName];
+          nicInfo.in = received.value + ' ' + received.units + '/s';
+          nicInfo.out = sent.value + ' ' + sent.units + '/s';
+
+          if (evt.data.sent_bytes - nicInfo.lastSent > 1024) {
+            nicInfo.lastSent = evt.data.sent_bytes;
+            this.tableService.updateStateInfoIcon(nicName, 'sent');
+          }
+
+          if (evt.data.received_bytes - nicInfo.lastReceived > 1024) {
```

Comment:
What is 1024? Some minimal limit? If so, let's introduce a variable to make it clear what's happening.